package/MDAnalysis/analysis/density.py

```diff
@@ -412,7 +420,6 @@ def __init__(self, atomgroup, delta=1.0,
         self._ydim = ydim
         self._zdim = zdim
 
-    def _prepare(self):
```

Comment:
This looks like the right change.

I am noticing that the necessary changes for parallelization really shift a lot of what `_prepare()` was doing back into `__init__`. I guess that's ok and it's better than introducing yet another method like `_prepare_init()` ... We probably need to review the docs for writing analysis code with AnalysisBase.

Reply:
Add a comment so that future readers know why we are doing everything here
and not in `_prepare()`.
```suggestion
        # The grid with its dimensions has to be set up in __init__
        # so that parallel analysis works correctly: each process
        # needs to have a `results._grid` of the same size and the
        # same `_bins`.
```

Reply:
manually added to the file, suggestions don't work on deleted lines